File: src/main/java/kookmin/software/capstone2023/timebank/application/service/account/AccountFinder.java
package kookmin.software.capstone2023.timebank.application.service.account;

import kookmin.software.capstone2023.timebank.domain.model.Account;
import kookmin.software.capstone2023.timebank.domain.repository.AccountJpaRepository;
import lombok.NoArgsConstructor;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.Optional;

@Service
@NoArgsConstructor
public class AccountFinder {
    private AccountJpaRepository accountJpaRepository;

    @Autowired
    public AccountFinder(AccountJpaRepository accountJpaRepository) {
        this.accountJpaRepository = accountJpaRepository;
    }

    public Account findById(Long accountId) {
        Optional<Account> ret = accountJpaRepository.findById(accountId);
        return ret.orElse(null);
    }
}

File: src/main/java/kookmin/software/capstone2023/timebank/application/service/account/AccountFinder.kt
package kookmin.software.capstone2023.timebank.application.service.account

import kookmin.software.capstone2023.timebank.domain.model.Account
import kookmin.software.capstone2023.timebank.domain.repository.AccountJpaRepository
import org.springframework.data.repository.findByIdOrNull
import org.springframework.stereotype.Service

@Service
// JPA 사용
// 레포지토리 개별 분리 필요
class AccountFinder(
    private val accountJpaRepository: AccountJpaRepository,
) {
    fun findById(accountId: Long): Account? {
        return accountJpaRepository.findByIdOrNull(accountId)
    }
}
